pkg/controller/volume/attachdetach/attach_detach_controller.go

```diff
@@ -293,6 +294,23 @@ func (adc *attachDetachController) nodeAdd(obj interface{}) {
 	// the attached volumes field. This function ensures that we sync with
 	// the actual status.
 	adc.actualStateOfWorld.SetNodeStatusUpdateNeeded(nodeName)
+
+	if _, exists := node.Annotations[volumehelper.ControllerManagedAttachAnnotation]; exists {
+		// Newly added node is managed by controller, ensure all pods are up to date.
+		adc.processPodsForNode(nodeName)
+	}
+}
+
+func (adc *attachDetachController) processPodsForNode(nodeName types.NodeName) {
+	pods, err := adc.podLister.List(labels.Everything())
```

Comment:
We could index the pod cache based on a NodeNameIndex and fetch pods for a specific node but I haven't looked close wrt the impact of using multiple indexes in cache. https://github.com/kubernetes/kubernetes/issues/38868